mkdocs/docs/index.md

```diff
@@ -1,22 +1,156 @@
 
-<!-- TODO(??): explain what bindings are, introduce options -->
+- [x] Linux
+- [x] Windows
+- [x] macOS, iOS
+- [x] Android
+- [x] Bare metal, including RISC-V
+- [x] ARM and x86
+- [ ] WebAssembly
+
+Accelerators and APIs
+
+- [x] Vulkan
+- [x] CUDA
+- [ ] Metal
+- [ ] WebGPU
+
+### Project architecture
+
+IREE adopts a _holistic_ approach towards ML model compilation: the IR produced
+contains both the _scheduling_ logic, required to communicate data dependencies
+to low-level parallel pipelined hardware/API like
+[Vulkan](https://www.khronos.org/vulkan/), and the _execution_ logic, encoding
+dense computation on the hardware in the form of hardware/API-specific binaries
+like [SPIR-V](https://www.khronos.org/spir/).
+
+<!-- TODO(scotttodd): text borders so this is easier to read on dark backgrounds -->
+<!-- TODO(scotttodd): use the same .svg as the README (point README at this path) -->
+![IREE Architecture](assets/images/iree_architecture.svg)
+
+## Workflow overview
+
+Using IREE involves these general steps:
+
+1. **Import your model**
+
+    Work in your framework of choice, then run your model through one of IREE's
+    import tools.
+
+2. **Select your deployment configuration**
+
+    Identify your target platform, accelerator(s), and other constraints.
+
+3. **Compile your model**
+
+    Compile through IREE, picking compilation targets based on your
+    deployment configuration.
+
+4. **Run your model**
+
+    Use IREE's runtime components to execute your compiled model.
+
+### Importing models from ML frameworks
+
+IREE supports importing models from multiple ML frameworks and model formats.
+
+<!-- TODO(#5555): rename "frontends" to "ML frameworks"? -->
+See the "Frontends" pages for guidance on working with each framework and
+model format.
+
+### Selecting deployment configurations
+
+IREE provides a flexible set of tools for various deployment scenarios - use as
+much or as little of IREE as you like.
+
+* What hardware should the bulk of your model run on? CPU? GPU?
+* What platforms are you targeting? Desktop? Mobile? An embedded system?
+* How fixed is your model itself? Can the weights be changed? Do you want
+  to support loading different model architectures dynamically?
+
+IREE supports the full set of these configurations using the same underlying
+technology.
+
+### Compiling models
+
+Model compilation is performed ahead-of-time on a _host_ machine for any
+combination of _targets_. The compilation process converts from "ops" used by
+high level frameworks down into optimized native code and associated scheduling
+logic.
+
+For example, compiling for GPU execution using Vulkan generates SPIR-V kernels
+and Vulkan API calls. For CPU execution, native code with static or dynamic
+linkage and the associated function calls are generated.
+
+### Running models
+
+IREE offers a low level C API, as well as several specialized sets of
+_bindings_ for running IREE models using other languages.
```

Comment:
Links to the section?

Reply:
Waiting to set up https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#section-index-pages for section linking to work well.